app/components/forms/wizard/attendee-step.js

```diff
@@ -23,16 +27,54 @@ export default Component.extend(FormMixin, {
         this.sendAction('save', this.data);
       });
     },
-    move(direction) {
+    savePublished() {
       this.onValid(() => {
-        this.sendAction('move', direction, this.data);
+        this.set('data.event.state', 'published');
+        this.sendAction('save', this.data);
       });
     },
-    publish() {
+    move(direction) {
       this.onValid(() => {
-        this.set('data.event.state', 'published');
-        this.sendAction('save', this.data);
+        this.sendAction('move', direction, this.data);
       });
+    },
+
+    openConfirmModal() {
+      this.set('isPublishUnpublishModalOpen', true);
+    },
+
+    togglePublishState() {
```

Comment:
Please move all common code to a shared instance after the changes are functionally approved. Same block of code should not be repeated anywhere